xrmocap/core/visualization/visualize_smpl.py

```diff
@@ -196,11 +210,13 @@ def visualize_smpl_data(
                 background_video, start=start_idx, end=end_idx)
         else:
             background_arr_batch = np.zeros(
-                shape=(end_idx - start_idx, cam_param.height, cam_param.width,
-                       3),
+                (end_idx - start_idx, cam_param.height, cam_param.width, 3),
                 dtype=np.uint8)
         batch_results = []
-        for frame_idx in range(start_idx, end_idx):
+
+        start_idx = 0
+        end_idx = min(data_len - curr_iter * batch_size, batch_size)
+        for frame_idx in tqdm(range(start_idx, end_idx), disable=disable_tqdm):
```

Comment:
What about we do not re-assign values of `start_idx` and `end_idx`, but edit for loop arguments:
```suggestion
        for relative_idx in tqdm(range(0, end_idx-start_idx), disable=disable_tqdm):
```